client/blocks/image-editor/index.jsx

```diff
@@ -95,6 +104,15 @@ const ImageEditor = React.createClass( {
 
 			this.setDefaultAspectRatio();
 		}
+
+		if ( newProps.originalAspectRatio && ! isEqual( newProps.originalAspectRatio, originalAspectRatio ) ) {
+			if ( newProps.originalAspectRatio.width > MinimumImageDimensions.WIDTH &&
+				newProps.originalAspectRatio.height > MinimumImageDimensions.HEIGHT ) {
+				this.setState( {
+					isGreaterThanMinimumDimensions: true
```

Comment:
Thanks @marcinbot and @gwwar . Good suggestion.

It did cross my mind, but I was ogling the image-editor-specific selectors, and they seemed pretty pure, so I assumed that sullying them with logic and constants imports wasn't the norm.

The originalAspectRatio values are set when `IMAGE_EDITOR_IMAGE_HAS_LOADED` is dispatched in `client/blocks/image-editor/image-editor-canvas`. The results we grab via the `getImageEditorOriginalAspectRatio` selector, but only after `image.onload()`.

So, at first glance, I'm wondering if it might be faster to grab the height/width values from the media prop handed down to `< ImageEditor />` as we originally did, and pass these to the new selector rather than waiting for the image to load and, therefore, `ui.editor.imageEditor.originalAspectRatio` to be assigned. (?)